hw_7/Solid_classes.py

```diff
@@ -0,0 +1,78 @@
+class Group:
+    def __init__(self):
+        self.group_n = list()
+
+    def add_to_group(self, obj):
+        self.group_n.append(obj)
+
+    def print_group_len(self):
+        return len(self.group_n)
+
+    def return_group(self):
+        return self.group_n
+
+
+class Person:
+    def __init__(self, name=None, age=None):
+        self.name = name
+        self.age = age
+
+    def __str__(self):
+        return self.__class__.__name__
+
+    def can_walk(self):
+        print(f'{self}, can walk..')
+
+    def can_run(self):
+        print(f'{self}, can run..')
+
+    def can_eat(self):
+        print(f'{self}, can eat..')
+
+    def can_sleep(self):
+        print(f'{self}, can sleep..')
+
+
+class Teacher(Person):
+    def __init__(self, name=None, age=None, stage=0):
+        super().__init__(name, age)
+        self.stage = stage
+
+    def can_teach(self):
+        print(f'{self}, can teach..')
+
+    def can_sleep(self):
+        super().can_sleep()
+        print('"Хоть не просыпайся..."\n')
+
+
+class Student(Person):
```

Comment:
this code is not following ISP (interface segregation principle)  we can run on the Student method teach what is forbidden.